packages/wrangler/src/r2/index.ts

```diff
@@ -51,6 +53,64 @@ export function r2(r2Yargs: CommonYargsArgv) {
 	return r2Yargs
 		.command("object", "Manage R2 objects", (r2ObjectYargs) => {
 			return r2ObjectYargs
+				.command(
+					"filter <bucket> <pattern>",
+					"Filter R2 objects in a bucket",
+					(filterArgs) => {
+						return filterArgs
+							.positional("bucket", {
+								describe: "The name of the bucket",
+								type: "string",
+							})
+							.positional("pattern", {
+								describe: "The pattern to test matches such as *.png",
```

Comment:
We should explicitly state that this is a glob pattern, I think

```suggestion
								describe: "The glob pattern to test objects against (ex: *.png, folder-*/**)",
```